url53f943c5cb_Praznat_AGP_tgz-pJ8-AMath_TestingJ8/src/GUI/TextDisplay/MemScroll.java

```diff
@@ -0,0 +1,31 @@
+package GUI.TextDisplay;
+
+import java.awt.Graphics;
+
+import Defs.M_;
+import Descriptions.BehName;
+import GUI.PopupAbstract;
+import Sentiens.*;
+import Sentiens.Law.PersonalCommandment;
+
+
+public class MemScroll extends Papyrus {
+	public MemScroll(PopupAbstract P) {
+		super(P);
+	}
+	public void paint(Graphics g) {
+		Clan clan = clan();
+		if(clan==null){return;}
+		super.paint(g);
+		int r = 0; String s;
+		for (M_ m : M_.values()) {
+			s = BehName.describeMem(clan, m);
+			if (s != "") {g.drawString(s, 2, 15+15*r++);}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.